viewing_party/party.py

```diff
@@ -1,23 +1,148 @@
 # ------------- WAVE 1 --------------------
 
+from symbol import subscriptlist
+
+
 def create_movie(title, genre, rating):
-    pass
+  new_move = {"title": title,
+              "genre" : genre,
+              "rating": rating}
+  for key, value in new_move.items():
+    if value == None:
+      return None
+  else:
+      return new_move
+
+def add_to_watched(user_data, movie):
+  user_data = {"watched":[movie]}
+  return user_data
+
+def add_to_watchlist(user_data,movie):
+  user_data = {"watchlist":[movie]}
+  return user_data
+
+def watch_movie(user_data, title):
+  for movie in user_data["watchlist"]:
+    if movie['title'] == title:
+      user_data["watched"].append(movie)
+      user_data["watchlist"].remove(movie)
+
+  return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+  movies_watched = user_data["watched"]
+  length = len(movies_watched)
+    
+  total = 0
 
+  for ratings in movies_watched:
+    total += float(ratings["rating"])
+    average = total / length
+  if length == 0:
+    return 0

```

Comment:
We can use the guard clause here and do this check earlier in the function since we don't want to execute lines 36-43 if the watched list is empty. 

On line 36, you can write:
```python 
if not user_data["watched"]: # recall that empty lists are Falsy
    return 0

# the rest of your logic follows the guard clause
```